### PR TITLE
Undeprecate interactiveshell

### PR DESCRIPTION
**Pull request by Carreau**
Rebase of #9575

### CODE REVIEW

File: IPython/terminal/interactiveshell.py [hunk 1]
```diff
@@ -1,21 +1,569 @@
-# -*- coding: utf-8 -*-
-"""DEPRECATED: old import location of TerminalInteractiveShell"""
-
-# Copyright (c) IPython Development Team.
-# Distributed under the terms of the Modified BSD License.
+"""IPython terminal interface using prompt_toolkit in place of readline"""
```
Comment (takluyver) [inline IPython/terminal/interactiveshell.py]: 'in place of readline' is redundant now that we no longer have an equivalent class using readline ;-)

File: docs/source/whatsnew/version5.rst [hunk 2]
```diff
@@ -34,12 +34,35 @@ that same event.
...
-IPython itself.
 
-the `PromptManager` class have been removed, and the prompt machinery simplified. 
-See `TerminalINteractiveShell.prompts` configurable for how to setup your prompts. 
+
+Replacement of readline in TerminalInteractiveShell and PDB
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+IPython 5.0 now uses ``prompt_toolkit``. The
+``IPtyhon.terminal.interactiveshell.TerminalInteractiveShell`` now uses
```
Comment (takluyver) [inline docs/source/whatsnew/version5.rst]: IPtyhon -> IPython

File: IPython/terminal/interactiveshell.py (hunk 1 above)
Comment (Carreau) [inline IPython/terminal/interactiveshell.py]: done.

File: docs/source/whatsnew/version5.rst (hunk 2 above)
Comment (Carreau) [inline docs/source/whatsnew/version5.rst]: done.

**Pull request merged**